editor/src/components/canvas/canvas-strategies/strategies/grid-resize-element-strategy.ts

```diff
@@ -0,0 +1,181 @@
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import * as EP from '../../../../core/shared/element-path'
+import type { GridElementProperties, GridPosition } from '../../../../core/shared/element-template'
+import { offsetPoint } from '../../../../core/shared/math-utils'
+import { assertNever } from '../../../../core/shared/utils'
+import { GridControls, GridResizeControls } from '../../controls/grid-controls'
+import { canvasPointToWindowPoint } from '../../dom-lookup'
+import type { CanvasStrategyFactory } from '../canvas-strategies'
+import { onlyFitWhenDraggingThisControl } from '../canvas-strategies'
+import type { InteractionCanvasState } from '../canvas-strategy-types'
+import {
+  getTargetPathsFromInteractionTarget,
+  emptyStrategyApplicationResult,
+  strategyApplicationResult,
+} from '../canvas-strategy-types'
+import type { InteractionSession } from '../interaction-state'
+import { getGridCellUnderMouse, setGridProps } from './grid-helpers'
+
+export const gridResizeElementStrategy: CanvasStrategyFactory = (
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession | null,
+  customState,
+) => {
+  const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+  if (selectedElements.length !== 1) {
+    return null
+  }
+
+  const selectedElement = selectedElements[0]
+  const isElementInsideGrid = MetadataUtils.isGridLayoutedContainer(
```

Comment:
you can use `MetadataUtils.isGridCell` here

Reply:
https://github.com/concrete-utopia/utopia/pull/6038/commits/6cd52afe7558d9ffa25a388d6f61357f54638d6e